system-addon/content-src/activity-stream-prerender.jsx

```diff
@@ -0,0 +1,63 @@
+const React = require("react");
+const ReactDOM = require("react-dom/server");
+const Base = require("content-src/components/Base/Base");
+const {Provider} = require("react-redux");
+const initStore = require("content-src/lib/init-store");
+const {reducers, INITIAL_STATE} = require("common/Reducers.jsm");
+const {actionTypes: at} = require("common/Actions.jsm");
+const allStrings = require("../data/locales.json");
+const EMPTY_LOCALE = "en-PRERENDER";
+
+function prerender(locale) {
+  const store = initStore(reducers, INITIAL_STATE);
+  let strings = {};
+
+  if (locale) {
+    if (!allStrings[locale]) {
+      throw new Error(`Tried to get strings for ${locale} but none were found.`);
+    }
+    strings = allStrings[locale];
+  } else {
+    Object.keys(allStrings["en-US"]).forEach(key => {
+      strings[key] = " ";
+    });
+  }
+
+  store.dispatch({
+    type: at.LOCALE_UPDATED,
+    data: {
+      locale: locale || EMPTY_LOCALE,
+      strings
+    }
+  });
+
+  store.dispatch({
+    type: at.PREFS_INITIAL_VALUES,
+    data: {
+      showTopSites: true,
+      showSearch: true,
+      topSitesCount: 6,
+      migrationExpired: true
+    }
+  });
+
+  store.dispatch({
+    type: at.SECTION_REGISTER,
+    data: {
+      enabled: true,
+      icon: "pocket",
+      id: "topstories",
+      order: 1,
+      title: {id: "header_recommended_by", values: {provider: "Pocket"}},
+      read_more_endpoint: "https://getpocket.cdn.mozilla.net/explore/trending?src=ff_new_tab",
+      topics: [{name: "Must Reads", category: "World", url: "https://getpocket.com/explore/must-reads?src=ff_new_tab"}]
+    }
+  });
+
```

Comment:
Since each of the above dispatches effectively replaces ones that currently come from chrome, but have a comment above each one talking about how they differ or any assumptions in the interest of future code maintainability.

Reply:
Particularly keeping in mind that effectively this code may also need to be updated as the chrome code changes over time.

Reply:
It's not really so much that it comes from chrome as that specific actions are needed to set properties in certain reducers

Reply:
I can actually add tests to help with this

Reply:
That sounds good.